File: 프로그래머스/1/340198. ［PCCE 기출문제］ 10번 ／ 공원/［PCCE 기출문제］ 10번 ／ 공원.py
def solution(mats, park):
    mats = sorted(mats)[::-1]
    for mat in mats:
        for m in range(len(park) - mat + 1):
            for n in range(len(park[m]) - mat + 1):
                inspector = []
                for i in range(mat):
                    for j in range(mat):
                        inspector.append(park[m + i][n + j])
            
                if inspector.count("-1") == mat ** 2:
                    return mat
        
    return -1

File: 프로그래머스/1/340198. ［PCCE 기출문제］ 10번 ／ 공원/［PCCE 기출문제］ 10번 ／ 공원.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> mats, vector<vector<string>> park) {
    sort(mats.begin(), mats.end(), greater<int>());

    int R = (int)park.size();
    int C = R ? (int)park[0].size() : 0;

    for (int mat : mats) {
        if (mat > R || mat > C) continue;  // ✅ 공원보다 크면 스킵

        for (int i = 0; i + mat <= R; i++) {
            for (int j = 0; j + mat <= C; j++) {

                bool ok = true;
                for (int m = 0; m < mat && ok; m++) {
                    for (int n = 0; n < mat; n++) {
                        if (park[i + m][j + n] != "-1") {
                            ok = false;
                            break;
                        }
                    }
                }

                if (ok) return mat;
            }
        }
    }
    return -1;
}
